src/iamclient/iamclient.cpp

```diff
@@ -190,10 +190,11 @@ aos::Error IAMClient::ApplyCertificate(const aos::String& nodeID, const aos::Str
 
     info.mCertURL = response.cert_url().c_str();
 
-    for (auto ch : response.serial()) {
-        if (auto err = info.mSerial.PushBack(ch); !err.IsNone()) {
-            return AOS_ERROR_WRAP(err);
-        }
+    auto err = aos::String(response.serial().c_str()).HexToByteArray(info.mSerial);
+    if (!err.IsNone()) {
+        LOG_ERR() << "Apply certificate failed to convert serial: err = " << err.Message();
```

Comment:
`Failed to convert serial` would be enough.

Reply:
Updated.